sphinx/builders/html/__init__.py

```diff
@@ -483,7 +505,8 @@ def prepare_writing(self, docnames: Set[str]) -> None:
             'parents': [],
             'logo': logo,
             'favicon': favicon,
-            'html5_doctype': html5_ready and not self.config.html4_writer
+            'html5_doctype': html5_ready and not self.config.html4_writer,
+            'pygments_dark_style': self.dark_highlighter is not None,
```

Comment:
How about calling `add_css_file()` API instead?
```
def init_highlighter(self):
    ...
    if self.dark_highlighter:
        self.add_css_file('pygments_dark.css', media="(prefers-color-scheme: dark)", id="pygments_dark_css")
```

Then you don't need to modify layout.html.

Reply:
I noticed there are no link tags for auxiliary styles.

Reply:
> How about calling `add_css_file()` API instead?

Used it in an earlier version though I think manually placing it in the layout file grants more control over the order of insertion if someone want to move pygments style files in their theme.

> I noticed there are no link tags for auxiliary styles.

You are right. I will refer to my previous comment:

> Additional pygments styles are generated for these options but not automatically included in the html. As this is a very niche option I think most people using it will benefit from the freedom this gives them.

Manual inclusion of these files would allow theme developers to e.g. postpone fetching of the stylesheets until the user actively decides to use these themes.

Reply:
If it is a very niche, -1 for adding. TBH, there are no enough resources for maintenance. So I'd like to add only important features to the core.

It is tough work to deprecate once published configurations, template variables, APIs and so on. So we should design carefully before adding them. Is the order of insertion really needed?

Reply:
Okay, so should I call `add_css_files()` for the `pygmnents_dark.css` file, the auxiliary styles or all of them? Or do you mean dropping complete support for auxiliary styles and only add support for a single additional dark pygments stylesheet?

Reply:
?

Reply:
I removed the link from the template and switched to using `add_css_file()` now

Reply:
Sorry for the late comment. I meant dropping complete support for auxiliary styles.